### DISCUSSION TITLE
[Proposal] Add `Add(IEnumerable items)` for `ICollection` to support the collection initialisation syntax

### DISCUSSION THREAD

**Discussion by MihailsKuzmins** [dotnet/runtime#50745] [General]
Would it be possible to add a method `Add(IEnumerable items)` for the `IList` interface?
My problem is that I would like to use the collection initialisation syntax on this model (unfortunately, the IList<Class> cannot be changed to the implementation).

```cs
public sealed class Class {}

public sealed class SomeClass
{
    public IList<Class> Classes { get; }
}

// this is what I want
var items = Enumerable.Empty<Class>();

var item = new SomeClass
{
    Classes = { items } // does not compile
};
```

I have tried creating an extension method, but it seems that generic extension methods for interfaces are not accepted as valid candidates for the collection initialisation 😭
```cs
public static void Add<T>(this IList<T> @this, IEnumerable<T> items)
{
    foreach (var item int items)
        @this.add(item);
}
```

So I thought maybe it could be possible to add this method to `IList<T>` or even `ICollection<T>`?

**Comment by HaloFour** [comment]
API suggestions are better made on the [Runtime repository](https://github.com/dotnet/runtime).  This repository is more specifically for the C# language itself, and the language wouldn't need any changes to call an `Add` instance or extension method if it were to exist.

**Reply by MihailsKuzmins** [reply]
@HaloFour, oh, didn't know that. Thank you. I will bear it in mind for the next time

**Comment by CyrusNajmabadi** [comment]
Transferring this to dotnet/runtime.  Thanks!

**Comment by huoyaoyuan** [comment]
Default interface methods makes modifying interface technically non-breaking, but not used in practice yet.
Supporting `AddRange` in collection initializer may not be a good idea. Similar has been complained in https://github.com/dotnet/csharplang/discussions/4522
